File: frontend/src/types/marketData.ts
/**
 * TypeScript interfaces mirroring the backend market-data schemas.
 * Display-layer types only.
 *
 * Source of truth: app/api/schemas/market_data.py
 */

export interface PricePoint {
  date: string
  open: number | null
  high: number | null
  low: number | null
  close: number | null
  volume: number | null
}

export interface PriceHistoryResponse {
  ticker: string
  period: string
  interval: string
  data_timestamp: string | null
  prices: PricePoint[]
}

File: app/api/schemas/market_data.py
"""
Response schemas for the market-data history endpoint.

Display/transport shapes only — no analysis or scoring logic. All numeric
fields are nullable so non-finite values (NaN/inf) can be safely serialized as
JSON ``null`` instead of producing invalid JSON.
"""

from __future__ import annotations

from pydantic import BaseModel


class PricePoint(BaseModel):
    """One OHLCV bar for a single date."""

    date: str
    open: float | None
    high: float | None
    low: float | None
    close: float | None
    volume: float | None


class PriceHistoryResponse(BaseModel):
    """Historical daily price series for one ticker."""

    ticker: str
    period: str
    interval: str
    data_timestamp: str | None
    prices: list[PricePoint]
